pennylane_forest/qvm.py

```diff
@@ -192,10 +203,10 @@ def apply_parametric_program(self, operations, **kwargs):
             # Prepare for parametric compilation
             par = []
             for param in operation.data:
-                if isinstance(param, Variable):
-                    # Using the idx for each Variable instance to specify the
+                if getattr(param, "requires_grad", False) and operation.name != "BasisState":
```

Comment:
The reason `BasisState` is disallowed, is we have a custom `basis_state` function in `device.py` that applies PauliX and Identity operations based on the state that is provided to `BasisState`.

It is unnecessary (and yields incorrect results) to then define a PyQuil memory reference for the array of states provided to `BasisState`, since we'd not have parametrized operations (only PauliX and Identity).

Reply:
👍